bundles/model/org.eclipse.smarthome.model.core/src/main/java/org/eclipse/smarthome/model/core/internal/ModelRepositoryImpl.java

```diff
@@ -119,11 +133,13 @@ public boolean addOrRefreshModel(String name, InputStream inputStream) {
                     }
                     notifyListeners(name, EventType.MODIFIED);
                     return true;
-                } catch (IOException e) {
-                    logger.warn("Configuration model '" + name + "' cannot be parsed correctly!", e);
-                    resourceSet.getResources().remove(resource);
                 }
             }
+        } catch (IOException e) {
+            logger.warn("Configuration model '" + name + "' cannot be parsed correctly!", e);
```

Comment:
Don't build strings in logging statements (I know it has been moved around only, but a good time to fix).